sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/collect.scala

```diff
@@ -139,6 +148,32 @@ case class CollectSet(
 
   def this(child: Expression) = this(child, 0, 0)
 
+  /*
+   * SPARK-31500
+   * Array[Byte](BinaryType) Scala equality don't works as expected
+   * so HashSet return duplicates, we have to change types to drop
+   * this duplicates and make collect_set work as expected for this
+   * data type
+   */
+  override lazy val bufferElementType = child.dataType match {
+    case BinaryType => ArrayType(BinaryType)
```

Comment:
ArrayType(ByteType)?

Reply:
Good catch, it works anyway but  I think it's better in this way